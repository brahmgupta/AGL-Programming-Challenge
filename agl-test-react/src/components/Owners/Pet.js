// @flow
import React from "react";

type Props = {
  petName: string
};

const Pet = ({ petName }: Props) => {
  return (
    <li data-testid="pet" className="pet"> {petName}</li>
  );
};

export default Pet;
