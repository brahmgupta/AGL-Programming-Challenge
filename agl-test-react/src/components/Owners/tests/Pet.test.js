import React from "react";
import { render } from "@testing-library/react";
import Pet from "../Pet";

const mockPetName = "Garfield";

describe("<Pet />", () => {
  it("renders the component", () => {
    const container = render(<Pet petName={mockPetName} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it("should render pets", () => {
    const { getByText } =  render(<Pet petName={mockPetName} />);
    const linkElement = getByText(/Garfield/i);
    expect(linkElement).toBeInTheDocument();
  });
});
