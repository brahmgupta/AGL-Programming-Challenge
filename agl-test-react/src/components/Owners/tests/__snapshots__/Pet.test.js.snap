// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Pet /> renders the component 1`] = `undefined`;
